File: NC/max-water-container.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:

        ans = 0

        left, right = 0, len(heights) - 1

        while left < right:
            ans = max(ans, (right-left)*min(heights[left], heights[right]))

            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1

        return ans

File: NC/max-water-container.cpp
class Solution
{
public:
    int maxArea(vector<int> &heights)
    {
        int ans = 0;
        for (int i = 0, j = heights.size() - 1; i < j;)
        {
            ans = max((j - i) * min(heights[i], heights[j]), ans);
            if (heights[i] < heights[j])
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return ans;
    }
};
